scrape/manager.go

```diff
@@ -54,8 +54,9 @@ type Manager struct {
 	targetsActive  []*Target
 	targetsDropped []*Target
 	targetsAll     map[string][]*Target
-	mtx            sync.RWMutex
 	graceShut      chan struct{}
+	mtxTargets     sync.RWMutex
+	mtx            sync.RWMutex
```

Comment:
Just use normal mutexes

Reply:
you are right about the global mutex , but the mtxTargets is accessed from many webhandler goroutines and should benefit from using a Read lock only. Am I wrong ?

Reply:
I doubt that the number of read is nigh enough to justify the additional complication. This isn't a hot path.